src/Interpreters/ReplaceQueryParameterVisitor.cpp

```diff
@@ -77,25 +77,26 @@ void ReplaceQueryParameterVisitor::visitQueryParameter(ASTPtr & ast)
 
 void ReplaceQueryParameterVisitor::visitIdentifier(ASTPtr & ast)
 {
-    auto & ast_identifier = ast->as<ASTIdentifier &>();
-    if (ast_identifier.children.empty())
+    auto ast_identifier = dynamic_pointer_cast<ASTIdentifier>(ast);
```

Comment:
typeid_cast or assert_cast?

Reply:
It's a shortcut - not to check `ASTIdentifier` and `ASTTableIdentifier` with the same code.